### ISSUE TITLE
Plan for releasing the models?

### ISSUE THREAD

**Issue by xiye17** [facebookresearch/coconut#3]
Thanks a lot for sharing the code. very impressive work. I wonder if you have plans for releasing the trained models so we can directly take it to run inference.

**Comment by Ber666** [comment]
Thank you for your interest in our work! Unfortunately we don't have the plan to release the trained models for now, since it would involve a much more complicated process.

**Comment by xiye17** [comment]
Got it thanks.

**Comment by wassname** [comment]
But has anyone managed to replicate the results? I couldn't.

Eziao with [open wandb logs](https://wandb.ai/weikaihuang-xidian-university/coconut?nw=nwusereziao) couldn't 

![Image](https://github.com/user-attachments/assets/e7a2be10-99db-460d-afe4-0009c28d6ce4)

**Comment by Ber666** [comment]
Hi, thanks for the question. I've confirmed that the exact code can reproduce the reported number. We used [4 GPUs](https://github.com/facebookresearch/coconut/blob/d1c10b134ad2bded92b0eca4c0a422c513a12dea/README.md?plain=1#L98) to train the model, which means the effective batch size is 32 * 4 = 128. In the wandb log you shared, the effective batch size seems to be 32?

**Comment by wassname** [comment]
Thanks for confirming. Huh, I wonder if that's the difference. It's plausible, but it would mean a large batch size is important for perhaps stability and regularisation?. Perhaps hobbyists with smaller setups should use gradient accumulation.

**Comment by Ber666** [comment]
I'm not sure if that's the only difference... I haven't tried using a smaller learning rate yet, so I can't say for sure. Gradient accumulation should definitely help for smaller setups though.

**Comment by wassname** [comment]
By the way, if you're cooking the next version, here are a few papers you might find useful:

- [Seq-VCR](https://github.com/rarefin/seq_vcr): they found that an extra loss term will stabilize the hidden states and let it think for longer. It might combine well with this project, in obvious ways
- "Suppressed neurons" are neurons that hide/suppress activations just before expression. It's likely that they contain the steering/thinking/metadata that should be passed to the next layer. That means that COCONUT might perform much better if using the 2nd or 3rd to last layer, rather than the last layer, which has lots of information "suppressed". More info in my repo: https://github.com/wassname/eliciting_suppressed_knowledge
- Explainability: COCONUT has [worried lots of people, including me](https://www.lesswrong.com/posts/D2Aa25eaEhdBNeEEy/worries-about-latent-reasoning-in-llms), because it's thinking silently, and it's harder to verify and debug its being honest. But if you add sparsity or a concept bottleneck, then you might actually get a really nice way to steer and intervene and inspect the model's thinking.
